lib/validation.js

```diff
@@ -81,25 +83,25 @@ function compileSchemasForValidation (context, compile, isCustom) {
     }
     context[headersSchema] = compile({ schema: headersSchemaLowerCase, method, url, httpPart: 'headers' })
   } else if (Object.hasOwnProperty.call(schema, 'headers')) {
-    throw new Error('headers schema is undefined')
+    warning.emit('FSTDEP015', 'headers', method, url)
   }
 
   if (schema.body) {
     context[bodySchema] = compile({ schema: schema.body, method, url, httpPart: 'body' })
   } else if (Object.hasOwnProperty.call(schema, 'body')) {
-    throw new Error('body schema is undefined')
+    warning.emit('FSTDEP015', 'body', method, url)
   }
 
   if (schema.querystring) {
     context[querystringSchema] = compile({ schema: schema.querystring, method, url, httpPart: 'querystring' })
   } else if (Object.hasOwnProperty.call(schema, 'querystring')) {
-    throw new Error('querystring schema is undefined')
+    warning.emit('FSTDEP015', 'querystring', method, url)
   }
 
   if (schema.params) {
     context[paramsSchema] = compile({ schema: schema.params, method, url, httpPart: 'params' })
   } else if (Object.hasOwnProperty.call(schema, 'params')) {
```

Comment:
This statement will also cover situations where the schema part is `null`, for instance not it won't match the sentence given on the deprecation.

wdyt if we extend the deprecation message to make it agnostic, and just indicate that the part is missed? 🤔

Reply:
Good catch! I've changed the message. See the example below. Any feedback is welcome :)
```console
[FSTDEP015] FastifyDeprecation: The params schema for GET: /:id is missed. This may indicate the schema is not well specified.
```